src/coreclr/vm/memberload.cpp

```diff
@@ -367,16 +367,23 @@ void MemberLoader::GetDescFromMemberRef(ModuleBase * pModule,
         {
             //
             // <NICE> this is duplicated logic GetFieldDescByIndex </NICE>
-            //
-            INDEBUG(mdFieldDef token = pFD->GetMemberDef();)
-
-            DWORD pos = static_cast<DWORD>(pFD - (pMT->GetApproxFieldDescListRaw() + pMT->GetNumIntroducedInstanceFields()));
-            _ASSERTE(pos >= 0 && pos < pMT->GetNumStaticFields());
+            //         
 
-            pFD = pMT->GetGenericsStaticFieldDescs() + pos;
-            _ASSERTE(pFD->GetMemberDef() == token);
-            _ASSERTE(!pFD->IsSharedByGenericInstantiations());
-            _ASSERTE(pFD->GetEnclosingMethodTable() == pMT);
+            DWORD pos = static_cast<DWORD>(pFD - (pMT->GetApproxFieldDescListRaw() + pMT->GetNumStaticFields()));
+            if(!(pos >= 0 && pos < pMT->GetNumStaticFields()))
```

Comment:
```suggestion
            if (pos >= pMT->GetNumStaticFields())
```

This can be simplified. The above is the same because `DWORD` is an unsigned integer, which is always positive or `0` so the first part is always true. We can then remove the negation and flip the `<` to `>=`.